org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java

```diff
@@ -0,0 +1,46 @@
+package org.code.javabuilder;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class InputRedirectionStreamTest {
+  private InputRedirectionStream stream;
+  private InputAdapter inputAdapter;
+
+  @BeforeEach
+  public void setup() {
+    inputAdapter = mock(InputAdapter.class);
+    when(inputAdapter.getNextMessage()).thenReturn("hello world");
+    stream = new InputRedirectionStream(inputAdapter);
```

Comment:
As another example, I would consider moving the setup code here into each function (or create a helper) so the arrange-act-assert pattern in each unit test clear.  Alternatively, you can instantiate `stream` and `inputAdapter` here in setup but at least the `when` line should go in each test (and be tailored to be only what is needed by the test).

Reply:
Thanks! That cleaned things up a bunch!